cmd/core/main.go

```diff
@@ -205,37 +207,44 @@ func main() {
 		os.Exit(1)
 	}
 
-	if err = (&plaincontrollers.BundleReconciler{
-		Client:     mgr.GetClient(),
-		Scheme:     mgr.GetScheme(),
-		Storage:    bundleStorage,
-		Finalizers: bundleFinalizers,
-		Unpacker:   unpacker,
-	}).SetupWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", rukpakv1alpha1.BundleKind)
+	commonBundleProvisionerOptions := []bundle.Option{
+		bundle.WithUnpacker(unpacker),
+		bundle.WithFinalizers(bundleFinalizers),
+		bundle.WithStorage(bundleStorage),
+	}
+
+	cfgGetter := helmclient.NewActionConfigGetter(mgr.GetConfig(), mgr.GetRESTMapper(), mgr.GetLogger())
+	acg := helmclient.NewActionClientGetter(cfgGetter)
+	commonBDProvisionerOptions := []bundledeployment.Option{
+		bundledeployment.WithReleaseNamespace(ns),
+		bundledeployment.WithActionClientGetter(acg),
+		bundledeployment.WithStorage(bundleStorage),
+	}
+
+	if err := bundle.SetupProvisioner(mgr, append(
+		commonBundleProvisionerOptions,
+		bundle.WithProvisionerID(plain.ProvisionerID),
+		bundle.WithHandler(bundle.HandlerFunc(plain.HandleBundle)),
+	)...); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", rukpakv1alpha1.BundleKind, "provisionerID", plain.ProvisionerID)
 		os.Exit(1)
 	}
 
-	if err = (&registrycontrollers.BundleReconciler{
-		Client:     mgr.GetClient(),
-		Scheme:     mgr.GetScheme(),
-		Storage:    bundleStorage,
-		Finalizers: bundleFinalizers,
-		Unpacker:   unpacker,
-	}).SetupWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", rukpakv1alpha1.BundleKind)
+	if err := bundle.SetupProvisioner(mgr, append(
+		commonBundleProvisionerOptions,
+		bundle.WithProvisionerID(registry.ProvisionerID),
+		bundle.WithHandler(bundle.HandlerFunc(registry.HandleBundle)),
```

Comment:
It took some time for me to understand this line.  Is this a popular way to pass an interface as an argument?

Reply:
This seems common with single-method interfaces. I think the most well-known one is `http.Handler`/`http.HandlerFunc`.